File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.go
func hasSameDigits(s string) bool {
    for len(s) > 2 {
        sub := []rune{} 
        for i := 0; i < len(s)-1; i++ {
            dig1 := s[i] - '0'
            dig2 := s[i+1] - '0'
            sum := (dig1 + dig2) % 10
            sub = append(sub, rune(sum+'0'))
        }

        s = string(sub)
    }    
    return s[0] == s[1]
}

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        int n = s.size();
        while(s.size() > 2) {
            string sub = "";
            for(int i=0;i+1<n;i++) {
                int d1 = s[i] - '0';
                int d2 = s[i + 1] - '0';
                int sum = (d1 + d2) % 10;
                sub.push_back(sum + '0');
            }
            s = sub;
            n = s.size();
        } 
        return s[0] == s[1];
    }
};